internal/metadatafile/dir_unix.go

```
@@ -0,0 +1,145 @@
+// Lstat describes name — a single path component — within d without
+// following a final symlink, using d's own file descriptor rather than a
+// fresh path walk. Unlike OpenFile/OpenChild, this never fails merely
+// because name is a symlink: the returned FileInfo describes the symlink
+// itself, exactly like the standard library's os.Lstat.
+func (d *Dir) Lstat(name string) (os.FileInfo, error) {
+	fd, err := unix.Openat(int(d.f.Fd()), name, unix.O_PATH|unix.O_NOFOLLOW|unix.O_CLOEXEC, 0)
+	if err != nil {
+		return nil, &os.PathError{Op: "openat", Path: filepath.Join(d.f.Name(), name), Err: err}
+	}
+	file := os.NewFile(uintptr(fd), filepath.Join(d.f.Name(), name))
+	defer file.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `file.Close` is not checked

(errcheck)